codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
@mach6 
The behavior kind of seems to be different between IDEs. eclipse shows it under a package named `default package` and Intellij doesn't do any of that. In either cases, the class definitely doesn't include a `package` declaration.

This makes sense to me here because I didn't provide a package name and the IDE defaulted to whatever was supposed to be its known default. But in our case, we are explicitly overriding the values with a value which is null or empty. So shouldn't the system just default to the edit checks in place to detect this rather than perform a recovery for the user ?
We do this in all other places such as our `Config` class wherein if the user explicitly overrides the browser flavor to an empty value, we try to use to and then throw an error no ? Why this change of behavior here ?

Another example of wherein we are going against the Maven conventions.
The maven surefire plugin lets us define the suite xml location via a properties. If I explicitly default it to an empty value, I guess I see a `NullPointerException` or something on those lines. 

So IMO we should use defaults if no values were provided. If values were provided we attempt to validate the values. If the validation fails, throw an error and bail out. If the values were valid, we just use them and move ahead.